### PR TITLE
Remove dead code from Data.HashMap.Array

### PR DESCRIPTION
**Pull request by sjakobi**
This was largely mechanical – does it remove anything that we'd prefer to keep?

Context: #270

### CODE REVIEW

**Comment by treeowl** [comment]
We can add some tests for the `Array` module, I suppose....

File: Data/HashMap/Array.hs [hunk 1]
```diff
@@ -571,49 +535,3 @@ traverse' f = \ !ary ->
...
---
---  traverse' (\x -> print x *> pure undefined) xs
---
--- will actually print all the values and then return undefined.
--- We could add a strict mapMWithIndex, operating in an arbitrary
--- Monad, that supported such rules, but we don't have that right now.
-{-# RULES
-"traverse/ST" forall f. traverse f = traverseST f
-"traverse/IO" forall f. traverse f = traverseIO f
- #-}
```
[the remark this answers is not in the document]
Comment (emilypi) [reply Data/HashMap/Array.hs:619]: correct me if i'm wrong, but these don't seem to be used at all in the project.

[the remark this answers is not in the document]
Comment (treeowl) [reply Data/HashMap/Array.hs:619]: Which two functions?

[the remark this answers is not in the document]
Comment (emilypi) [reply Data/HashMap/Array.hs:619]: Deleted - n/m. i was mistaken.

[the remark this answers is not in the document]
Comment (sjakobi) [reply Data/HashMap/Array.hs:619]: Oops! Will revert.

I've made https://github.com/haskell-unordered-containers/unordered-containers/issues/272 to track this.

**Comment by sjakobi** [comment]
@treeowl I think I addressed your comments. Could you take another look?

This also still needs to be squashed before merging.

**Comment by treeowl** [comment]
Great.

**Pull request merged**